### PR TITLE
Fix and improve sample rate timing

### PR DESCRIPTION
**Pull request by JonaMata**
- Increased sample rate timer precision to allow for more specific sample rates
- Improved sample rate timing code to negate accumulating error

### CODE REVIEW

File: esp32_bluetooth_host/esp32_bluetooth_host.ino [hunk 1]
```diff
@@ -93,6 +94,10 @@ class SettingsUpdatedCallback : public BLECharacteristicCallbacks {
     int reqFrequency = std::max(std::min(value, (uint16_t)MAX_SAMPLING_FREQUENCY), (uint16_t)MIN_SAMPLING_FREQUENCY);
     samplingDelay = round((float)TIMER_PRECISION / (float)reqFrequency);
     frequency = TIMER_PRECISION / samplingDelay;
+    Serial.print("Sampling delay:\t");
+    Serial.println(samplingDelay);
+    Serial.print("Sampling frequency:\t");
+    Serial.println(frequency);
```
Comment (FrankBosman) [inline esp32_bluetooth_host/esp32_bluetooth_host.ino:108]: Maybe remove the debugging code, or make it more clear that it is to stay with a comment

File: esp32_bluetooth_host/esp32_bluetooth_host.ino
Comment (FrankBosman) [inline esp32_bluetooth_host/esp32_bluetooth_host.ino:1]: Update the version number

**Review by FrankBosman — CHANGES_REQUESTED** [review]
We should also first merge the blinking LED and merge it in this one so we don't have to flash them twice

File: esp32_bluetooth_host/esp32_bluetooth_host.ino
Comment (JonaMata) [reply esp32_bluetooth_host/esp32_bluetooth_host.ino:1]: I'll do this when everything is merged and we're ready for deploying a new version

**Review by FrankBosman — CHANGES_REQUESTED** [review]
Could you update the version number and update the branch with main?

**Comment by MaxLiebe** [comment]
Frank already mentioned my comments, LGTM otherwise

**Review by FrankBosman — APPROVED** [review]

**Pull request merged**